cpp/collections_module/algorithms/algorithms.cpp

```diff
@@ -0,0 +1,41 @@
+#include "algorithms.hpp"
+
+void Collections::RemoveAll(mgp_list *args, mgp_graph *memgraph_graph, mgp_result *result, mgp_memory *memory) {
+  mgp::memory = memory;
+  const auto arguments = mgp::List(args);
+  const auto record_factory = mgp::RecordFactory(result);
+
+  try {
+    const auto input_list = arguments[0].ValueList();
+    const auto to_remove_list = arguments[1].ValueList();
+    std::set<mgp::Value> searchable;
+
+    for (const auto value : input_list) {
+      searchable.insert(std::move(value));
+    }
```

Comment:
you can join these two into one part:
`std::set<mgp::Value> searchable{input_list.begin(), input_list.end()};`

Reply:
i tried this but it won't compile and I'm not sure I understand why:
error: no matching function for call to ‘std::_Rb_tree<mgp::Value, mgp::Value, std::_Identity<mgp::Value>, std::less<mgp::Value>, std::allocator<mgp::Value> >::_M_insert_range_equal(mgp::List::Iterator&, mgp::List::Iterator&)’

Reply:
Leave it as it is